components/calendar/src/persian.rs

```diff
@@ -0,0 +1,960 @@
+//!
+//! ```rust
+//! use icu::calendar::{Date, DateTime};
+//!
+//! // `Date` type
+//! let persian_date = Date::try_new_persian_date(1348, 10, 11)
+//!     .expect("Failed to initialize Persian Date instance.");
+//!
+//! // `DateTime` type
+//! let persian_datetime = DateTime::try_new_persian_datetime(1348, 10, 11, 13, 1, 0)
+//!     .expect("Failed to initialize Persian DateTime instance.");
+//!
+//! // `Date` checks
+//! assert_eq!(persian_date.year().number, 1348);
+//! assert_eq!(persian_date.month().ordinal, 10);
+//! assert_eq!(persian_date.day_of_month().0, 11);
+//!
+//! // `DateTime` checks
+//! assert_eq!(persian_datetime.date.year().number, 1348);
+//! assert_eq!(persian_datetime.date.month().ordinal, 10);
+//! assert_eq!(persian_datetime.date.day_of_month().0, 11);
+//! assert_eq!(persian_datetime.time.hour.number(), 13);
+//! assert_eq!(persian_datetime.time.minute.number(), 1);
+//! assert_eq!(persian_datetime.time.second.number(), 0);
+//! ```
+
+use crate::any_calendar::AnyCalendarKind;
+use crate::calendar_arithmetic::{ArithmeticDate, CalendarArithmetic};
+use crate::helpers::{ceil_div, div_rem_euclid64, i64_to_i32, I32Result};
+use crate::iso::Iso;
+use crate::julian::Julian;
+use crate::rata_die::RataDie;
+use crate::{types, Calendar, CalendarError, Date, DateDuration, DateDurationUnit, DateTime};
+use ::tinystr::tinystr;
+use core::marker::PhantomData;
+
+const FIXED_PERSIAN_EPOCH: RataDie = Julian::fixed_from_julian(ArithmeticDate {
+    year: (622),
+    month: (3),
+    day: (19),
+    marker: core::marker::PhantomData,
+});
+/// The Persian Calendar
+///
+/// The [Persian Calendar] is a solar calendar used officially by the countries of Iran and Afghanistan and many Persian-speaking regions.
+/// It has 12 months and other similarities to the Gregorian Calendar
+///
+/// This type can be used with [`Date`] or [`DateTime`] to represent dates in this calendar.
+///
+/// [Persian Calendar]: https://en.wikipedia.org/wiki/Solar_Hijri_calendar
+///
+/// # Era codes
+/// This calendar supports only one era code, which starts from the year of the Hijra, designated as "ah".
+#[derive(Copy, Clone, Debug, Default, Hash, Eq, PartialEq, PartialOrd, Ord)]
+#[allow(clippy::exhaustive_structs)]
+pub struct Persian;
+
+#[derive(Copy, Clone, Debug, Hash, Eq, PartialEq, PartialOrd, Ord)]
+
+/// The inner date type used for representing [`Date`]s of [`Persian`]. See [`Date`] and [`Persian`] for more details.
+pub struct PersianDateInner(ArithmeticDate<Persian>);
+
+impl CalendarArithmetic for Persian {
+    fn month_days(year: i32, month: u8) -> u8 {
+        match month {
+            1 | 2 | 3 | 4 | 5 | 6 => 31,
+            7 | 8 | 9 | 10 | 11 => 30,
+            12 if Self::is_leap_year(year) => 30,
+            12 => 29,
+            _ => 0,
+        }
+    }
+
+    fn months_for_every_year(_: i32) -> u8 {
+        12
+    }
+
+    fn is_leap_year(p_year: i32) -> bool {
```

Comment:
nit: link to source for the math (the lisp code?)

can also mention book section but ideally include a link to the code